jscomp/runtime/js_weakset.ml

```diff
@@ -1,3 +1,8 @@
 (** ES6 WeakSet API *)
 
 type 'a t
+
+external make : unit -> 'a t = "WeakSet" [@@mel.new]
+external add : value:'a -> 'a t = "add" [@@mel.send.pipe: 'a t]
```

Comment:
btw this is not 100% safe because`value` must be an object. Trying to add a primitive value like an `int` fails.

Reply:
so should we make it take a `_ Js.t` or a `_ Js.Dict.t` instead?

Reply:
I don't know. If I'm not mistaken, this means it would not be possible to pass normal OCaml record while I guess it should be ok since they compile to JS objects. See example [`1097608` (#1058)](https://github.com/melange-re/melange/pull/1058/commits/109760840e02088551842462ea42e59a6412103b)

Reply:
that's fair, but at least with `_ Js.t` we can know that it's exactly a JS object.

To me, the tradeoff here seems like: 

- if you keep `'a`, you can pass any argument to the functions in the module, at the cost of runtime errors when you pass non-object arguments
- if you enforce `_ Js.t`, then you lose the ability to pass some well-known-to-work types (such as records, which are nominal, and you can't make a function accept "all records")


Personally, I'd prefer if we shipped 99% runtime-free, correct-by-construction code. IMO it's also pretty easy to change it downstream:

- `include` the `Js.WeakMap` module and overwrite the externals you wish to change signatures
- use `external toJsDotT: myRecord -> _ Js.t = "%identity"` to convert your records / other types that compile to objects at the boundary so that you can pass them to `Js.WeakMap`.

What do you think?

Reply:
I couldn't use `Js.t` because the `Js` module is not accessible from `jscomp/runtime` and I couldn't find an alternative in `Js_*`. So I ended up using `Js_dict.t`.

Reply:
I think that's a fair compromise.